tests/unit/test_stats_reporter.py

```diff
@@ -279,3 +341,108 @@ def test_draw_histogram_invalid_input():
+        "\t< 214.954KB: 6 ▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇▇\n"
+        "\t< 407.531KB: 0 \n"
+        "\t< 772.638KB: 0 \n"
+        "\t<=1.431MB  : 2 ▇▇▇▇▇▇▇▇▇\n"
+        "\t----------------------------------------\n"
+        "\tmax: 1.431MB\n"
+        "\n"
+        "📂 [bold]Allocator type distribution:[/]\n"
+        "\t 1: 1013\n"
+        "\t 4: 797\n"
+        "\t 3: 152\n"
+        "\t 10: 4\n"
+        "\n"
+        "🥇 [bold]Top 5 largest allocating locations (by size):[/]\n"
+        "\t- fake_func:fake.py:5 -> 5.000MB\n"
+        "\t- fake_func2:fake.py:10 -> 3.000KB\n"
+        "\t- __main__:fake.py:15 -> 4.000B\n"
+        "\n"
+        "🥇 [bold]Top 5 largest allocating locations (by number of allocations):[/]\n"
+        "\t- fake_func:fake.py:5 -> 20\n"
+        "\t- fake_func2:fake.py:10 -> 50\n"
+        "\t- __main__:fake.py:15 -> 1"
+    )
+    if sys.version_info[0] == 3 and sys.version_info[1] == 7:
+        printed = "\n".join(" ".join(x[0]) for x in mocked_print.call_args_list)
+    else:
+        printed = "\n".join(" ".join(x.args) for x in mocked_print.call_args_list)
+    assert expected == printed
+
+
+def test_stats_output_json(fake_stats, tmp_path):
+    reporter = StatsReporter(fake_stats, 5)
+    with patch("json.dump") as json_dump:
+        reporter.render(to_json=True, result_path=tmp_path)
+    expected = {
+        "total_num_allocations": 20,
+        "total_bytes_allocated": 3341500,
+        "allocation_size_histogram": [
+            (4739, 1),
+            (8986, 0),
+            (17036, 3),
+            (32299, 0),
+            (61237, 1),
+            (116099, 1),
+            (220113, 6),
+            (417312, 0),
+            (791181, 0),
+            (1500000, 2),
+        ],
+        "allocator_type_distrbution": [
+            ("MALLOC", 1013),
+            ("REALLOC", 797),
+            ("CALLOC", 152),
+            ("MMAP", 4),
+        ],
+        "top_allocations_by_size": [
+            {"location": "fake_func:fake.py:5", "size": 5242880},
+            {"location": "fake_func2:fake.py:10", "size": 3072},
+            {"location": "__main__:fake.py:15", "size": 4},
+        ],
+        "top_allocations_by_count": [
+            {"location": "fake_func:fake.py:5", "count": 20},
+            {"location": "fake_func2:fake.py:10", "count": 50},
+            {"location": "__main__:fake.py:15", "count": 1},
+        ],
+        "metadata": {
+            "start_time": "2023-01-01 01:00:00",
+            "end_time": "2023-01-01 02:00:00",
+            "total_allocations": 3341500,
+            "total_frames": 10,
+            "peak_memory": 1500000,
+            "command_line": "fake stats",
+            "pid": 123456,
+            "python_allocator": "pymalloc",
+            "has_native_traces": False,
+        },
+    }
+    if sys.version_info[0] == 3 and sys.version_info[1] == 7:
+        actual = json_dump.call_args[0][0]
+    else:
+        actual = json_dump.call_args.args[0]
```

Comment:
Same as above, I think we could unconditionally do:
```suggestion
    actual = json_dump.call_args[0][0]
```